### PR TITLE
Disable LeakSanitizer build

### PR DESCRIPTION
**Pull request by kripken**
It started to fail consistently for no apparent reason, so probably a change on CI infrastructure.

Example log: https://travis-ci.org/WebAssembly/binaryen/jobs/326886345

Example log with the extra logging it recommends: https://travis-ci.org/WebAssembly/binaryen/jobs/326886066

Any ideas? If not, we can disable it for now, if it's a CI thing it might start working again later.

### CODE REVIEW

**Comment by dschuff** [comment]
No ideas, but you should disable it the same way we did in https://github.com/WebAssembly/binaryen/pull/183 so we still get ASan without leak detection.

**Comment by binji** [comment]
Yeah, UBSan started failing on the wabt bots, since I was pulling clang-3.6 and travis recently updated to clang-5.0. I was using `ubuntu-toolchain-r-test` too, perhaps that broke it.

That said, recent runs seem to be fixed on wabt ubsan, so maybe the Travis infra folks fixed it.

**Comment by dschuff** [comment]
We could match wabt and switch from the waterfall clang (which we haven't updated in forever) to travis's clang (e.g. https://github.com/WebAssembly/binaryen/commit/8ff8e68db558e0fc5de923a7fb5baa991e49f822) but that would be orthogonal to this change I think.

**Comment by kripken** [comment]
Sadly updating the compiler didn't help. I added a commit to disable just the leak checks in a proper way as suggested above.

File: .travis.yml [hunk 1]
```diff
@@ -28,34 +26,26 @@ jobs:
...
-        CXX_COMPILER="./test/wasm-install/wasm-install/bin/clang++"
         COMPILER_FLAGS="-fsanitize=undefined -fno-sanitize-recover=all -fsanitize-blacklist=$(pwd)/ubsan.blacklist"
 
+    # FIXME we currently must disable leak detection, see #1351
     - <<: *test-ubuntu
       env: |
-        CC_COMPILER="./test/wasm-install/wasm-install/bin/clang"
-        CXX_COMPILER="./test/wasm-install/wasm-install/bin/clang++"
-        COMPILER_FLAGS="-fsanitize=address"
+        COMPILER_FLAGS="-fsanitize=address" ASAN_OPTIONS="detect_leaks=0"
```
Comment (dschuff) [inline .travis.yml]: Do you know how this interacts with line 18  above? I guess this must override that so maybe we should add symbolize=1 here too.

Comment (kripken) [reply .travis.yml]: Good catch. I can just move this to there I guess, seems safer for future changes.

Comment (kripken) [reply .travis.yml]: But nevermind, this didn't fix it anyhow. LSAN is just crashing entirely, not in a way affected by disabling the leak check feature. Pushed a commit to do it the original way again.

**Pull request merged**